.github/workflows/solidity-foundry.yml

```diff
@@ -47,127 +119,247 @@ jobs:
+          || contains(fromJson(needs.changes.outputs.all_changes), 'shared')
+          || needs.changes.outputs.non_src_changes == 'true')
+          && matrix.product.setup.run-coverage }}
         run: |
           sudo apt-get install lcov
-          ./contracts/scripts/ccip_lcov_prune ./contracts/lcov.info ./lcov.info.pruned
+          ./contracts/scripts/lcov_prune ${{ matrix.product.name }} ./contracts/lcov.info ./contracts/lcov.info.pruned
 
-      - name: Report code coverage
-        if: ${{ contains(fromJson('["ccip"]'), matrix.product) && needs.changes.outputs.changes == 'true' }}
+      - name: Report code coverage for ${{ matrix.product.name }}
+        if: ${{ (contains(fromJson(needs.changes.outputs.all_changes), matrix.product.name)
+          || contains(fromJson(needs.changes.outputs.all_changes), 'shared')
+          || needs.changes.outputs.non_src_changes == 'true')
+          && matrix.product.setup.run-coverage }}
         uses: zgosalvez/github-actions-report-lcov@a546f89a65a0cdcd82a92ae8d65e74d450ff3fbc # v4.1.4
         with:
-          update-comment: true
-          coverage-files: lcov.info.pruned
-          minimum-coverage: 98.5
-          artifact-name: code-coverage-report
+          update-comment: false
+          coverage-files: ./contracts/lcov.info.pruned
+          minimum-coverage: ${{ matrix.product.min-coverage }}
+          artifact-name: code-coverage-report-${{ matrix.product.name }}
           working-directory: ./contracts
-          github-token: ${{ secrets.GITHUB_TOKEN }}
 
       - name: Collect Metrics
-        if: needs.changes.outputs.changes == 'true'
+        if: ${{ contains(fromJson(needs.changes.outputs.all_changes), matrix.product.name)
+          || contains(fromJson(needs.changes.outputs.all_changes), 'shared')
+          || needs.changes.outputs.non_src_changes == 'true' }}
         id: collect-gha-metrics
         uses: smartcontractkit/push-gha-metrics-action@d9da21a2747016b3e13de58c7d4115a3d5c97935 # v3.0.1
         with:
-          id: solidity-foundry
+          id: ${{ matrix.product.name }}-solidity-foundry
+          org-id: ${{ secrets.GRAFANA_INTERNAL_TENANT_ID }}
+          basic-auth: ${{ secrets.GRAFANA_INTERNAL_BASIC_AUTH }}
+          hostname: ${{ secrets.GRAFANA_INTERNAL_HOST }}
+          this-job-name: Foundry Tests ${{ matrix.product.name }}
+        continue-on-error: true
+
+  analyze:
+    needs: [ changes ]
+    name: Run static analysis
+    if: needs.changes.outputs.not_test_sol_modified == 'true'
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout the repo
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+        with:
+          submodules: recursive
+
+      - name: Setup NodeJS
+        uses: ./.github/actions/setup-nodejs
+
+      - name: Install semver
+        shell: bash
+        run: |
+          npm install -g semver
```

Comment:
I believe the previous `setup-nodejs` action will install the dependencies from `./package.json` which already includes this `semver` package. Can probably remove this step.